snapcraft/internal/sources/_git.py

```diff
@@ -40,33 +40,39 @@ def __init__(self, source, source_dir, source_tag=None, source_commit=None,
                 'can\'t specify both source-branch and source-commit for '
                 'a git source')
 
-    def pull(self):
-        if os.path.exists(os.path.join(self.source_dir, '.git')):
-            refspec = 'HEAD'
-            if self.source_branch:
-                refspec = 'refs/heads/' + self.source_branch
-            elif self.source_tag:
-                refspec = 'refs/tags/' + self.source_tag
-            elif self.source_commit:
-                refspec = self.source_commit
+    def _pull_existing(self):
+        refspec = 'HEAD'
+        if self.source_branch:
+            refspec = 'refs/heads/' + self.source_branch
+        elif self.source_tag:
+            refspec = 'refs/tags/' + self.source_tag
+        elif self.source_commit:
+            refspec = self.source_commit
 
-            # Pull changes to this repository and any submodules.
-            subprocess.check_call([self.command, '-C', self.source_dir,
-                                   'pull', '--recurse-submodules=yes',
-                                   self.source, refspec])
+        # Pull changes to this repository and any submodules.
+        subprocess.check_call([self.command, '-C', self.source_dir,
+                               'pull', '--recurse-submodules=yes',
+                               self.source, refspec])
+
+        # Merge any updates for the submodules (if any).
+        subprocess.check_call([self.command, '-C', self.source_dir,
+                              'submodule', 'update'])
+
+    def _clone_new(self):
+        command = [self.command, 'clone', '--recursive']
```

Comment:
Isn't that what '--recursive' is?  From the 'git log' history I didn't change that line.